services/api/kalvi/middlewares/AdminUserMiddleware.py

```diff
@@ -0,0 +1,30 @@
+# custom_middleware.py
+from rest_framework import status
+from django.http import JsonResponse
+from rest_framework_simplejwt.tokens import UntypedToken
+from db.models import User
+from rest_framework_simplejwt.exceptions import InvalidToken, TokenError
+
+class AdminUserMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+    def __call__(self, request):
+        if 'HTTP_AUTHORIZATION' in request.META:
+            try:
+                # Extract token and retrieve user object with better error handling
+                token = UntypedToken(request.META['HTTP_AUTHORIZATION'].split()[1])
```

Comment:
request.headers.get('Authorization'): This syntax directly accesses the headers of the request using the headers attribute of the request object. It tries to get the value of the 'Authorization' header, and if the header is not present, it returns None.

request.META['HTTP_AUTHORIZATION']: This syntax accesses the request's META dictionary, which contains all available HTTP headers. The 'HTTP_AUTHORIZATION' key corresponds to the Authorization header. If the 'Authorization' header is not present, accessing request.META['HTTP_AUTHORIZATION'] directly will raise a KeyError. since i had provided if clause to check that it exists already so i didn't see harm in accessing it direct without request object. 
although since your consideration for uniformity i'll access it through request object everywhere.